File: missing-number-in-arithmetic-progression/missing-number-in-arithmetic-progression.swift
class Solution {
    func missingNumber(_ arr: [Int]) -> Int {
        return (arr.first! + arr.last!) * (arr.count + 1) / 2 - arr.reduce(0, +)
    }
}

//         (arr.first! + arr.last!)*(arr.count+1)/2 - arr.reduce(0, +)

/// Hello

File: missing-number-in-arithmetic-progression/missing-number-in-arithmetic-progression.py
class Solution:
    def missingNumber(self, arr: List[int]) -> int:
        
        diff = int((arr[-1] - arr[0]) / len(arr))
                        
        for idx in range(1, len(arr)):
            if arr[idx] - arr[idx-1] != diff:
                return arr[idx-1] + diff        
        
        return arr[-1]